commons-executors/src/main/java/com/palantir/common/concurrent/ExecutorInheritableThreadLocal.java

```diff
@@ -166,6 +162,19 @@ protected T initialValue() {
         return new MapMaker().weakKeys().concurrencyLevel(1).makeMap();
     }
 
+    private static Object wrapNull(Object value) {
+        return value == null ? new NullWrapper() : value;
+    }
+
+    @SuppressWarnings("unchecked")
+    private T unwrapNull(Object ret) {
+        if (ret instanceof NullWrapper) {
```

Comment:
Once we use enum for NullWrapper, this can be identity check for `ret == NullWrapper.INSTANCE`